build-template/build.sh

```diff
@@ -1,7 +1,7 @@
-#!/bin/bash
+#!/bin/sh
 
 if test `uname` = Darwin; then
-    cachedir=~/Library/Caches/KBuild
+    cachedir=$HOME/Library/Caches/KBuild
```

Comment:
I understand this enables the script to use `/bin/sh`. But why make this one inconsistent with most others in the system? Don't think we have any plans to switch from `bash` in general. (May have a few `sh` scripts for test coverage or historical reasons.)